hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/TargetFilterAddUpdateLayout.java

```diff
@@ -135,9 +157,46 @@ public ComponentContainer getRootComponent() {
         return filterQueryLayout;
     }
 
+    protected void setCSVExportCallback(final ProxyTargetFilterQuery proxyEntity){
+        StreamResource csvResource = getStreamResourceToCSV(proxyEntity);
+        fileDownloader = new FileDownloader(csvResource);
+        fileDownloader.extend(exportButton);
+    }
+
+    protected void removeDownloaderExtension(){
+        if(fileDownloader != null){
+            fileDownloader.remove();
+        }
+    }
+
+    private StreamResource getStreamResourceToCSV(final ProxyTargetFilterQuery proxyEntity){
+
+        final StreamSource streamSource = () -> {
+            final PipedInputStream in = new PipedInputStream();
+
+            try {
+                final PipedOutputStream out = new PipedOutputStream(in);
+                final TargetDataCsvExporter csvExporter = new TargetDataCsvExporter(targetManagement, ";",
+                        "E MMM dd HH:mm:ss z yyyy");
+
+                SpringContextHolder.getInstance().getBean("uiExecutor", Executor.class)
+                        .execute(() -> csvExporter.writeTargetsByFilterId(proxyEntity.getId(), out));
```

Comment:
I would rather use `csvExporter.writeTargetsByFilterId(proxyEntity.getQuery(), out));`, see the comment above